### PR TITLE
Adding requirements.txt with esphome dependency.

### PR DESCRIPTION
**Pull request by barndawgie**

### CODE REVIEW

**Comment by tronikos** [comment]
What's the reason for this? How will it be used? I've personally created a python virtual environment where I have only pip installed esphome and pre-commit. Given you only have one dependency, does having requirements.txt help?

**Comment by barndawgie** [comment]
> What's the reason for this? How will it be used? I've personally created a python virtual environment where I have only pip installed esphome and pre-commit. Given you only have one dependency, does having requirements.txt help?

If you checkout the code and open it in VS, it will prompt you to install requirements when e.g. creating a new Virtual Environment. That was my primary thinking - helping new contributors with hints to get their environment setup correctly.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+esphome>=2023.2.0
+pre-commit>=3.4.0
+yamllint>=1.32.0
```
Comment (tronikos) [inline requirements.txt:3]: This was added in pre-commit so no reason to additionally require it.

Comment (barndawgie) [reply requirements.txt]: Done.

**Pull request merged**